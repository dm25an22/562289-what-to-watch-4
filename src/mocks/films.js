
const COUNT_FILMS = 1;

const text = `Lorem Ipsum is simply dummy text of the printing and typesetting industry.
 Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer
  took a galley of type and scrambled it to make a type specimen book.`;

const films = [
  {
    title: `Fantastic Beasts`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Dramas`
  },
  {
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Revenant`,
    src: `img/revenant.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Johnny English`,
    src: `img/johnny-english.jpg`,
    genre: `Kids & Family`
  },
  {
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Thrillers`
  },
  {
    title: `Seven Years in Tibet`,
    src: `img/seven-years-in-tibet.jpg`,
    genre: `Documentary`
  },
  {
    title: `Snatch`,
    src: `img/snatch.jpg`,
    genre: `Romance`
  },
  {
    title: `Mindhunter`,
    src: `img/mindhunter.jpg`,
    genre: `Horror`
  },
  {
    title: `Pulp Fiction`,
    src: `img/pulp-fiction.jpg`,
    genre: `Crime`
  },
  {
    title: `What we do in the shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`,
    genre: `Sci-Fi`
  },
  {
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Comedies`
  }
];

const actorsList = [
  `Robert Agnew`,
  `Jonathan Ahdout`,
  `Waris Ahluwalia`,
  `Ahmed Ahmed`,
  `Philip Ahn`,
  `Charles Aidman`,
  `Danny Aiello`,
  `Danny Aiello`,
  `Liam Aiken`,
  `Alan Aisenberg`,
  `Spottiswoode Aitken`,
  `Franklyn Ajaye`,
  `Ayad Akhtar`,
  `Gbenga Akinnagbe`,
  `Claude Akins`,
  `Marc Alaimo`,
  `Mark Alan`,
  `Matthew Alan`,
  `Rico Alaniz`,
  `Carlos Alazraqui`,
  `Luis Alberni`,
  `Devin Albert`,
  `Eddie Albert`,
];

const producers = [
  `Edward Janney`,
  `Sabra Jones`,
  `Tom Ruegger`,
  `Ron Senkowski`,
  `Kelly Leonard`,
  `Matthew Daniels`,
  `Olivia Luccardi`,
  `P. J. Wolfson`,
  `Yance Ford`,
  `Allen Estrin`,
  `Jenno Topping`,
];

const previews = [
  `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
];

const getRandomNumber = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

const getRandomRating = () => {
  const firstNumber = getRandomNumber(0, 10);
  return Number(`${firstNumber}.${firstNumber !== 10 && firstNumber !== 0 ? getRandomNumber(0, 9) : 0}`);
};

const getDescriptionRating = (rating) => {

  switch (true) {
    case rating < 3:
      return `Bad`;

    case rating >= 3 && rating < 5:
      return `Normal`;

    case rating >= 5 && rating < 8:
      return `Good`;

    case rating >= 8 && rating < 10:
      return `Very good`;

    case rating >= 10:
      return `Awesome`;
  }

  return `Unknown`;
};

const getRandomItem = (arr) => {
  return arr[getRandomNumber(0, arr.length - 1)];
};

const getRandomArrWithUniqeItems = (arr) => {
  const copyArr = [...arr];
  const result = [];
  for (let i = 0; i < getRandomNumber(2, arr.length - 1); i++) {
    result.push(copyArr.splice(getRandomNumber(0, copyArr.length - 1), 1)[0]);
  }

  return result;
};


const generateMock = () => {
  const ratingNumber = getRandomRating();
  const ratingString = getDescriptionRating(ratingNumber);
  const randomFilm = getRandomItem(films);

  return {
    title: randomFilm.title,
    moviePoster: `https://picsum.photos/300/450`,
    bigPoster: `https://picsum.photos/1900/550`,
    smallCardImg: randomFilm.src,
    genre: randomFilm.genre,
    year: getRandomNumber(1995, 2017),
    description: text,
    preview: getRandomItem(previews),
    rating: ratingNumber,
    descriptionRating: ratingString,
    quantityVotes: getRandomNumber(5, 300),
    producer: getRandomItem(producers),
    listActors: getRandomArrWithUniqeItems(actorsList)
  };
};

const createMock = (countFilms) => {
  return new Array(countFilms)
    .fill()
    .map(generateMock);
};

export default createMock(COUNT_FILMS);

