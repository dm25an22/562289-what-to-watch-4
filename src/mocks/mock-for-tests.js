const mockFilms = [
  {
    title: `Fantastic Beasts`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 1
  },
  {
    title: `Title`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Drama`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 2
  },
  {
    title: `Title2`,
    moviePoster: `img/the-grand-budapest-hotel-poster.jpg`,
    bigPoster: `img/the-grand-budapest-hotel-poster.jpg`,
    genre: `Action`,
    year: 2011,
    description: `Something`,
    rating: 4.7,
    descriptionRating: `Normal`,
    quantityVotes: 125,
    producer: `Tom Ruegger`,
    listActors: [`one`, `two`],
    smallCardImg: `img/macbeth.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    backgroundColor: `red`,
    runTime: 90,
    id: 3
  },
];

const mockPromo = {
  title: `Promo`,
  moviePoster: `film.poster_image`,
  bigPoster: `film.background_image`,
  smallCardImg: `film.preview_image`,
  genre: `film.genre`,
  year: 1998,
  description: `film.description`,
  preview: `film.preview_video_link`,
  rating: 3.2,
  descriptionRating: `Bad`,
  quantityVotes: 1290,
  producer: `film.director`,
  listActors: [`1`, `2`, `3`],
  runTime: 120,
  videoLink: `film.video_link`,
  isFavorite: true,
  backgroundColor: `red`,
  id: 4
};

export {
  mockFilms,
  mockPromo
};
