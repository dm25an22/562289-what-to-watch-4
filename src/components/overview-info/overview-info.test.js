import React from "react";
import renderer from "react-test-renderer";
import OverviewInfo from "./overview-info.jsx";
import {mockFilms} from "../../mocks/mock-for-tests";

it(`render OverviewInfo component`, () => {
  const tree = renderer.create(
      <OverviewInfo
        film={mockFilms[1]}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
