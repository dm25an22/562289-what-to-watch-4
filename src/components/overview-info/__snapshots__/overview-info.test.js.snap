// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render OverviewInfo component 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      4,7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        125
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Something
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Tom Ruegger
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        one, two
      </strong>
    </p>
  </div>,
]
`;
